### ISSUE TITLE
Test DB not created when running "rake db:setup"

### ISSUE THREAD

**Issue by benthorner** [alphagov/govuk-docker#452]
This isn't specific to GOV.UK Docker, but seems to affect all repos running Rails 6.1+. A workaround for this is to run:

```
govuk-docker-run env RAILS_ENV=test bundle exec rake db:setup
```

While it's something of a regression, it appears to be intentional and related to [this Rails issue](https://github.com/rails/rails/issues/27299). Next steps:

- Find the change in Rails that lead to this change in behaviour. Hopefully this will lead to documentation about the recommended approach, or we could refer to [the mainstream docs](https://guides.rubyonrails.org/v3.2/testing.html).

- Agree the fix and apply this to each affected app. For example, we may agree to add the above workaround to each `Makefile`. Ideally we should preserve the old behaviour, as that was simpler.

**Comment by ChrisBAshton** [comment]
Did a little digging earlier, I think these two links might be relevant:
- https://github.com/rails/rails/pull/39027
- https://github.com/rails/rails/commit/6ca9031ba3c389f71366c3e6abf069c6924c5acf

**Comment by ChrisBAshton** [comment]
Just tried the steps in the mainstream docs. `rake db:test:prepare` in [step 1](https://guides.rubyonrails.org/v3.2/testing.html#the-three-environments), as well as `rake db:migrate` and `rake db:test:load` in [the preparation section](https://guides.rubyonrails.org/v3.2/testing.html#preparing-your-application-for-testing) all fail with `ActiveRecord::NoDatabaseError: Unknown database 'contacts_test'`.

**Comment by benthorner** [comment]
Thanks for investigating @ChrisBAshton! I looked a bit deeper and found that it's due to [this PR](https://github.com/rails/rails/pull/37351). It should only affect people running apps with GOV.UK Docker, as opposed to directly on their machine.

Some options:

1. We could move away from `DATABASE_URL` entirely. This is used by the deployed versions of our apps, so it would be quite a difficult change to make overall.

2. We could manually set the `url` for the development environment in each app, so that it reads from e.g. `ENV["DEV_DATABASE_URL"]`, much like we do for test ([here](https://github.com/alphagov/email-alert-api/blob/59e9ec81ac1caee1c85e6109d9b28a9e8a96045c/config/database.yml#L20)).

3. We could live with the regression and add the workaround to each `Makefile`.

I think (3) makes the most sense, as this is only an issue when running apps via GOV.UK Docker. It's also the easiest option. It just won't help people who run `db:create` commands manually in GOV.UK Docker.

Shall we go with that?

**Comment by ChrisBAshton** [comment]
I've just tried option 2 locally with Contacts Admin and it works nicely. This feels like the best option if we ignore the upfront work involved, based on my understanding that everything would 'just work' as it did before. I.e. people can `db:create` to their hearts' content. The long term gain of not having to remember - and support other people in remembering - might outweigh the short term pain of the work involved.

That work would, in my head, be:

1) Duplicate the `DATABASE_URL` for each app in govuk-docker:

```diff
      DATABASE_URL: "mysql2://root:root@mysql-5.5/contacts_development"
+     DEV_DATABASE_URL: "mysql2://root:root@mysql-5.5/contacts_development"
```

2) Work our way through all the apps to [specify the database URL](https://github.com/alphagov/email-alert-api/blob/59e9ec81ac1caee1c85e6109d9b28a9e8a96045c/config/database.yml#L20) for the dev environment.

3) Then a tidy-up PR to remove `DATABASE_URL`

4) Also we'd have to tell people to run `rake db:create` for the changes to take effect

Does that sound about right? In which case, step 2 is the painful bit, and step 4 is arguably not much better than the reality Option 3 presents. I think given Rails' convention over configuration, going out of our way to rename the DB URL to something non-standard also doesn't feel like a great idea.

...so I think I've rounded back onto Option 3 as the best way forward 🤔 but hopefully helpful to have these alternatives in writing!
